src/api/forms/repositories/aggregation/types.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @typedef {object} FilterConditions
+ * @property {{ $regex: RegExp }} [title] - Optional MongoDB regex query for title matching
+ */
+
+/**
+ * @typedef {object} RegexMatch
+ * @property {string} input - Input string to match against
+ * @property {string} regex - Regular expression pattern
+ * @property {string} options - Regex options (e.g., 'i' for case-insensitive)
+ */
+
+/**
+ * @typedef {object} SwitchCase
+ * @property {[{ $toLower: string } | string, string]} [$eq] - Equality comparison condition
+ * @property {RegexMatch} [$regexMatch] - Regular expression matching condition
+ */
+
+/**
+ * @typedef {object} SwitchBranch
+ * @property {SwitchCase} case - Condition to evaluate
+ * @property {number} then - Value to return if condition is true
+ */
+
+/**
+ * @typedef {object} SwitchExpression
+ * @property {Array<SwitchBranch>} branches - Array of condition-value pairs
```

Comment:
Not sure why it doesn't do it in JSDoc (maybe you know?)

But I remember in TypeScript it'd suggest this change 🤷‍♂️ 

```suggestion
 * @property {SwitchBranch[]} branches - Array of condition-value pairs
```

Reply:
Yeah, force of habit.